capstone/src/main/webapp/script.js

```diff
@@ -12,48 +12,104 @@
+ * Creates a Material card holding restaurant data and adds it to an outer HTML
+ * element
+ * @param restaurant  the subject restaurant
+ * @param containerElement  the HTML element that will hold the restaurant
  */
-function createRestaurantElement(restaurant) {
-  // Article tag to encapsulate restaurant elements
-  const restaurantElement = document.createElement('h2');
-  const linkElement = document.createElement('a');
+function addRestaurant(restaurant, containerElement) {
+  const restaurantDiv = document.createElement('div');
+  restaurantDiv.classList.add('mdc-card', 'restaurant-container');
 
-  // Restaurant element UI details
-  const redirect =
-      '/restaurantDetails.html?restaurantKey=' + restaurant.restaurantKey;
-  linkElement.setAttribute('href', redirect);
+  const restaurantNameDiv = createRestaurantNameDiv(restaurant);
+  const restaurantCuisineDiv = createRestaurantCuisineDiv(restaurant);
+  const restaurantStrugglingDiv = createRestaurantIsStrugglingDiv(restaurant);
+
+  restaurantDiv.appendChild(restaurantNameDiv);
+  restaurantDiv.appendChild(restaurantCuisineDiv);
+  restaurantDiv.appendChild(restaurantStrugglingDiv);
+
+  containerElement.appendChild(restaurantDiv);
+}
+
+/**
+ * Creates a div containing the name of a given restaurant
+ * @param restaurant  the subject restaurant
+ * @return {HTMLDivElement} div containing a link to a restaurant's details page
+ */
+function createRestaurantNameDiv(restaurant) {
+  const params = new URLSearchParams();
+  params.append('restaurantKey', restaurant.restaurantKey);
+
+  const linkElement = document.createElement('a');
+  linkElement.classList.add('restaurant-name-link');
   linkElement.innerText = restaurant.name;
-  restaurantElement.appendChild(linkElement);
+  linkElement.href = '/restaurantDetails?' + params;
+
+  const restaurantNameHeader = document.createElement('h3');
+  restaurantNameHeader.appendChild(linkElement);
+
+  const restaurantNameDiv = document.createElement('div');
+  restaurantNameDiv.classList.add(
+      'restaurant-name-container', 'mdc-typography--headline6');
+  restaurantNameDiv.appendChild(restaurantNameHeader);
+
+  return restaurantNameDiv;
+}
+
+/**
+ * Creates a div containing a list of buttons representing restaurant cuisines
+ * @param restaurant  the subject restaurant
+ * @return {HTMLDivElement} div holding all of the cuisine buttons
+ */
+function createRestaurantCuisineDiv(restaurant) {
+  const restaurantCuisineDiv = document.createElement('div');
+  restaurantCuisineDiv.classList.add('restaurant-cuisine-container');
+  restaurant.cuisine.forEach((item) => {
+    const cuisineItemButton = document.createElement('button');
+    cuisineItemButton.innerHTML = item;
+    cuisineItemButton.classList.add('mdc-button', 'mdc-ripple-surface');
+    restaurantCuisineDiv.appendChild(cuisineItemButton);
+  });
+
+  return restaurantCuisineDiv;
+}
+
+/**
+ * Creates a dive that displays a message based on whether a restaurant is
```

Comment:
div?

Reply:
Yeah, haha. Thanks.

Reply:
I made that mistake a bunch of times while writing the doc comments, but I thought I caught them all.